File: 2021_random_set_self_practice/1904.cpp
/*
input :
4

output :
5
*/

#include <iostream>
#include <algorithm>
using namespace std;

// 전역 변수 선언
int N;

int main(){
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    freopen("input.txt", "r", stdin);

    cin >> N;
    if (N == 1) {
        cout << 1 << '\n';
        return 0;
    }
    if (N == 2) {
        cout << 2 << '\n';
        return 0;
    }

    int a = 1;
    int b = 2;
    int tmp;
    int c;
    while (N - 2 != 0){
        tmp = b;

        c = (a + b) % 15746;
        b = c;
        a = tmp;

        N--;
    }
    cout << c << '\n';

    return 0;
}

File: 2021_random_set_self_practice/using_python/1904.py
"""
input :
4

output :
5
"""
import sys

sys.stdin = open("input.txt", "r")
input = sys.stdin.readline
MOD = 15746

if __name__ == "__main__":
    N = int(input())

    a = 1
    b = 2
    if N == 1:
        print(1)
        exit(0)
    if N == 2:
        print(2)
        exit(0)

    while N - 2:
        c = (a + b) % MOD
        b, a = c, b
        N -= 1

    print(c)
